sklearn/decomposition/_factor_analysis.py

```diff
@@ -159,6 +161,18 @@ class FactorAnalysis(_ClassNamePrefixFeaturesOutMixin, TransformerMixin, BaseEst
     (1797, 7)
     """
 
+    _parameter_constraints = {
+        "n_components": [Interval(Integral, 0, None, closed="left"), None],
+        "tol": [Interval(Real, 0.0, None, closed="left")],
+        "copy": ["boolean"],
+        "max_iter": [Interval(Integral, 1, None, closed="left")],
+        "noise_variance_init": ["array-like", None],
```

Comment:
you're right that ``noise_variance_init`` can be an array-like and not just a numpy.ndarray. Can you update the parameter description in the docstring ?
``noise_variance_init : array-like of shape (n_features,), default=None``